stingray/modeling/gpmodeling.py

```diff
@@ -0,0 +1,850 @@
+        for i in params_list:
+            if isinstance(prior_dict[i], tfpd.Distribution):
+                parameter = yield Prior(prior_dict[i], name=i)
+            else:
+                parameter = yield prior_dict[i]
+            prior_list.append(parameter)
+        return tuple(prior_list)
+
+    return prior_model
+
+
+def get_likelihood(params_list, kernel_type, mean_type, **kwargs):
+    """
+    A log likelihood generator function based on given values.
+    Makes a jaxns specific log likelihood function which takes in the
+    parameters in the order of the parameters list, and calculates the
+    log likelihood of the data given the parameters, and the model
+    (kernel, mean) of the GP model. **Note** Any parameters with a prefix
+    of "log_" are taken to be in the log scale.
+
+    Parameters
+    ----------
+    params_list:
+        A list in order of the parameters to be used.
+
+    prior_dict:
+        A dictionary of the priors of parameters to be used.
+
+    kernel_type:
+        The type of kernel to be used in the model.
+
+    mean_type:
+        The type of mean to be used in the model.
+
+    **kwargs:
+        The keyword arguments to be used in the log likelihood function.
+        **Note**: The keyword arguments Times and counts are necessary for
+        calculating the log likelihood.
+        Times: np.array or jnp.array
+            The time array of the lightcurve
+        counts: np.array or jnp.array
+            The photon counts array of the lightcurve
+
+    Returns
+    -------
+    The jaxns specific log likelihood function.
+
+    """
+    if not jax_avail:
+        raise ImportError("Jax is required")
+
+    if not can_make_gp:
+        raise ImportError("Tinygp is required to make the GP model.")
+
+    @jit
+    def likelihood_model(*args):
+        dict = {}
+        for i, params in enumerate(params_list):
+            if params[0:4] == "log_":
+                dict[params[4:]] = jnp.exp(args[i])
+            else:
+                dict[params] = args[i]
+        kernel = get_kernel(kernel_type=kernel_type, kernel_params=dict)
+        mean = get_mean(mean_type=mean_type, mean_params=dict)
+        gp = GaussianProcess(kernel, kwargs["Times"], mean_value=mean(kwargs["Times"]))
+        return gp.log_probability(kwargs["counts"])
+
+    return likelihood_model
+
+
+class GPResult:
+    """
+    Makes a GPResult object which takes in a Stingray.Lightcurve and samples parameters of a model
+    (Gaussian Process) based on the given prior and log_likelihood function.
+
+    Parameters
+    ----------
+    lc: Stingray.Lightcurve object
+        The lightcurve on which the bayesian inference is to be done
+
+    Other Parameters
+    ----------------
+    time : class: np.array
+        The array containing the times of the lightcurve
+
+    counts : class: np.array
+        The array containing the photon counts of the lightcurve
+
+    """
+
+    def __init__(self, Lc: Lightcurve) -> None:
```

Comment:
`Lc` should be all lowercase. Note: we generally use uppercase characters only in class definitions and almost nowhere else.

Reply:
Done